eng/pipelines/templates/jobs/ci.yml

```diff
@@ -226,17 +228,39 @@ jobs:
           BaselineFilePath: $(Build.sourcesdirectory)/eng/dotnet.gdnbaselines
           ServiceDirectory: ${{ parameters.ServiceDirectory }}
 
-  - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
-    parameters:
-      JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
-      MatrixConfigs: ${{ parameters.MatrixConfigs }}
-      MatrixFilters: ${{ parameters.MatrixFilters }}
-      MatrixReplace: ${{ parameters.MatrixReplace }}
-      CloudConfig:
-        Cloud: Public
-      AdditionalParameters:
-        SDKType: ${{ parameters.SDKType }}
-        ServiceToTest: ${{ coalesce(parameters.ServiceToTest, parameters.ServiceDirectory) }}
-        ServiceDirectory: ${{ parameters.ServiceDirectory }}
-        TestSetupSteps: ${{ parameters.TestSetupSteps }}
-        TestTimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }}
+        ${{ if ne(parameters.TestDependsOnDependency, 'not-specified') }}:
+          - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
+            parameters:
+              JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
+              MatrixConfigs: ${{ parameters.MatrixConfigs }}
+              MatrixFilters: ${{ parameters.MatrixFilters }}
+              MatrixReplace: ${{ parameters.MatrixReplace }}
+              CloudConfig:
+                Cloud: public
+              SparseCheckoutPaths: 
+                - "!/sdk/*/**/SessionRecords"
+              PreGenerationSteps: 
+              - template: /eng/pipelines/templates/steps/dependency.tests.yml
+                parameters:
+                  TestDependsOnDependency: ${{ parameters.TestDependsOnDependency }}
+                  ProjectListOverrideFileProperty: 'ProjectListOverrideFileProperty'
+              AdditionalParameters:
+                SDKType: ${{ parameters.SDKType }}
+                ServiceDirectory: ${{ parameters.ServiceDirectory }}
+                TestSetupSteps: ${{ parameters.TestSetupSteps }}
+                TestTimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }}
+                ProjectListOverrideFileProperty: 'ProjectListOverrideFileProperty'
+        ${{ if ne(parameters.ServiceDirectory, '') }}:
```

Comment:
We should not hide this behind empty service directory, because it could generate an invalid template which would be hard to debug.

Reply:
I would either merge it with the above section as I mentioned in my above comment, or just use `{{ else }}:`